models/Analysis.ts

```diff
@@ -0,0 +1,136 @@
+import parseISO from 'date-fns/fp/parseISO'
+
+import { ApiData } from '@/api/ApiData'
+
+/**
+ * An analysis recommendation.
+ */
+class Recommendation {
+  /**
+   * @param endExperiment A boolean value indicating whether the experiment should end (based only on the raw data and
+   *        ignoring any warnings).
+   * @param chosenVariationId The ID of the variation that should be implemented if the experiment should end.
+   *        This field is `null` if the experiment should continue running.
+   * @param reason The reason for the recommendation. See `RecommendationReason`.
+   * @param warnings An array of string warnings. See `RecommendationWarning`.
+   */
+  // istanbul ignore next (skip coverage for auto-generated constructor)
+  constructor(
+    public readonly endExperiment: boolean,
+    public readonly chosenVariationId: number | null,
+    public readonly reason: RecommendationReason,
+    public readonly warnings: Array<RecommendationWarning>,
+  ) {}
+
+  /**
+   * Create an instance from raw API data (parsed JSON).
+   *
+   * @param apiData Raw API data.
+   */
+  static fromApiData(apiData: ApiData): Recommendation {
+    return {
+      endExperiment: apiData.end_experiment,
+      chosenVariationId: apiData.chosen_variation_id,
+      reason: apiData.reason as RecommendationReason,
+      warnings: apiData.warnings.map((warning: string) => warning as RecommendationWarning),
```

Comment:
If it's possible for `apiData.warnings` to be null:

```suggestion
      warnings: apiData.warnings?.map((warning: string) => warning as RecommendationWarning),
```

Reply:
It isn't. :slightly_smiling_face: